robots/pkg/flakefinder/downloader.go

```diff
@@ -130,6 +130,66 @@ func findUnitTestFileForJob(ctx context.Context, client *storage.Client, bucket
 	return reports, nil
 }
 
+func FindUnitTestFilesForPeriodicJob(ctx context.Context, client *storage.Client, bucket string, periodicJobDir []string, startOfReport time.Time, skipBeforeStartOfReport bool) ([]*JobResult, error) {
+
+	dirOfJobs := path.Join(periodicJobDir...)
+
+	jobDirs, err := ListGcsObjects(ctx, client, bucket, dirOfJobs+"/", "/")
+	if err != nil {
+		return nil, fmt.Errorf("error listing gcs objects: %v", err)
+	}
+	builds := sortBuilds(jobDirs)
+
+	profilePath := ""
+	buildNumber := 0
+	reports := []*JobResult{}
+	for _, build := range builds {
+		buildDirPath := path.Join(dirOfJobs, strconv.Itoa(build))
+		dirOfFinishedJSON := path.Join(buildDirPath, finishedJSON)
+
+		// Fetch file attributes to check whether this test result should be included into the report
+		attrsOfFinishedJsonFile, err := ReadGcsObjectAttrs(ctx, client, bucket, dirOfFinishedJSON)
+		if err == storage.ErrObjectNotExist {
+			// build still running?
+			continue
+		} else if err != nil {
+			return nil, err
+		}
+		isBeforeStartOfReport := attrsOfFinishedJsonFile.Created.Before(startOfReport)
+		if skipBeforeStartOfReport && isBeforeStartOfReport {
+			logrus.Infof("Skipping test results before %v for %s in bucket '%s'\n", startOfReport, buildDirPath, bucket)
```

Comment:
How many lines does this produce? Will it print out all periodics which are too old? If yes, that would be really a lot.

Reply:
Never mind, there is a `break` below, not a `continue`.